src/vm.rs

```diff
@@ -59,48 +66,43 @@ impl VirtualMachine {
             ),
             settings,
             stack_pool,
+            panic: Box::new(Instruction::from_panic(
```

Comment:
Dumb question: Would extending `ExecutionStatus` to have an additional "panicked" variant significantly degrade performance? Or would it interfere with plans of tail call optimizations?

Reply:
Right now `ExecutionStatus::Running` is performing a tail call and `ExecutionStatus::Stopped` is reaching a base case. I'm not sure how the new case would translate.

However, I think it is possible to implement panic more cleanly by doing something like directly calling it.